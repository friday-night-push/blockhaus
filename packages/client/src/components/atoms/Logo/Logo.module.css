.logo {
  display: flex;
}

.logo[data-size='normal'] {
  width: 340px;
}

.logo[data-size='small'] {
  width: 240px;
}
