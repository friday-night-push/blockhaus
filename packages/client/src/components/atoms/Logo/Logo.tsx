import logoWithBlocks from 'src/assets/logos/blockhaus-full.svg';
import logo from 'src/assets/logos/blockhaus.svg';

import styles from './Logo.module.css';

export type LogoProps = {
  size?: 'auto' | 'sm' | 'md';
  isFull?: boolean;
};

export const Logo = ({ size = 'md', isFull = false }: LogoProps) => {
  return (
    <img
      className={styles.logo}
      data-size={size}
      src={isFull ? logoWithBlocks : logo}
      alt="Blockhaus"
    />
  );
};
