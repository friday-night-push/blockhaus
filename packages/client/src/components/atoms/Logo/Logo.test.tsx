import { render } from '@testing-library/react';

import { Logo } from './Logo';

describe('Logo', () => {
  it('renders correctly with default props', () => {
    const { container } = render(<Logo />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders with size sm', () => {
    const { getByAltText } = render(<Logo size="sm" />);
    const img = getByAltText('Blockhaus');
    expect(img).toHaveStyle('width: 240px');
  });

  it('renders with size "auto"', () => {
    const { getByAltText } = render(<Logo size="auto" />);
    const img = getByAltText('Blockhaus');
    expect(img).toHaveStyle('width: auto');
  });
});
