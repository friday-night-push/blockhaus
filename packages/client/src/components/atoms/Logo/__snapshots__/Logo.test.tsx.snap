// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo renders correctly with default props 1`] = `
<img
  alt="Blockhaus"
  style="display: flex; width: 340px;"
/>
`;
